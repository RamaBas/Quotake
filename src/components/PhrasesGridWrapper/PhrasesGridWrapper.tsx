import React from 'react';
import { Grid, Box, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import PhraseCard from '../PhraseCard/PhraseCard';

interface PhrasesGridWrapperProps {
  phrases: Array<{
    id: string;
    text: string;
  }>;
  onDelete?: (id: string) => void;
  onRestore?: (id: string) => void; // Falta generar pantalla de eliminados
  emptyMessage?: string;
  cardVariant?: 'default' | 'deleted';
}
const PhrasesGridWrapper: React.FC<PhrasesGridWrapperProps> = ({
  phrases,
  onDelete,
  onRestore,
  emptyMessage,
  cardVariant = 'default'
}) => {
  const { t } = useTranslation();

  if (phrases.length === 0) {
    return (
      <Box sx={{ 
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: '200px'
      }}
      data-testid="empty-state"
      >
        <Typography variant="h6" color="textSecondary">
          {emptyMessage || t('noPhrasesFound')}
        </Typography>
      </Box>
    );
  }

  return (
    <Grid container spacing={2} sx={{ mt: 2 }} data-testid="phrases-grid">
      {phrases.map((phrase) => (
        /* Tuve algunos problemas con la libreria de MUI y Typescript, por lo que tuve que usar un @ts-ignore 
        para permitirme deployar en la version productiva en Vercel */
        // @ts-ignore
        <Grid sm={6} md={4} lg={3} key={phrase.id}>
          <PhraseCard
            phrase={phrase}
            onDelete={onDelete && onDelete}
            onRestore={onRestore && onRestore}
            variant={cardVariant}
          />
        </Grid>
      ))}
    </Grid>
  );
};

export default PhrasesGridWrapper;