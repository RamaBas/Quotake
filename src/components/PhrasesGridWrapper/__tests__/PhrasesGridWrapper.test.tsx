import { render, screen } from '@testing-library/react';
import PhrasesGridWrapper from '../PhrasesGridWrapper';

describe('PhrasesGridWrapper Component', () => {
  const mockPhrases = [
    { id: '1', text: 'Frase de prueba 1' },
    { id: '2', text: 'Frase de prueba 2' },
  ];

  it('renders grid with phrases correctly', () => {
    const { container } = render(
      <PhrasesGridWrapper 
        phrases={mockPhrases} 
        onDelete={jest.fn()} 
      />
    );
    
    const grid = screen.getByTestId('phrases-grid');
    expect(grid).toBeInTheDocument();
    
    // Snapshot test
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders empty state with default message', () => {
    render(
      <PhrasesGridWrapper phrases={[]} />
    );
    
    expect(screen.getByText('noPhrasesFound')).toBeInTheDocument();

  });

  it('renders empty state with custom message', () => {
    const customMessage = 'No se encontraron resultados';
    render(
      <PhrasesGridWrapper 
        phrases={[]} 
        emptyMessage={customMessage}
      />
    );

    expect(screen.getByText(customMessage)).toBeInTheDocument();
    expect(screen.queryByText('noPhrasesFound')).not.toBeInTheDocument();

  });
});