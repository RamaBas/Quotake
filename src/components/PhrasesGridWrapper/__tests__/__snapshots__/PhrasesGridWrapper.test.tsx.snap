// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhrasesGridWrapper Component renders grid with phrases correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-xi2iqh-MuiGrid-root"
  data-testid="phrases-grid"
>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-152yur6-MuiPaper-root-MuiCard-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-oy3rny"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fqlg31-MuiSvgIcon-root"
          data-testid="FormatQuoteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
          />
        </svg>
      </div>
      <div
        class="MuiCardContent-root css-rnega1-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1salclq-MuiTypography-root"
        >
          Frase de prueba 1
        </p>
      </div>
      <button
        aria-label="Eliminar frase"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-2wse5v-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-152yur6-MuiPaper-root-MuiCard-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-oy3rny"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fqlg31-MuiSvgIcon-root"
          data-testid="FormatQuoteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
          />
        </svg>
      </div>
      <div
        class="MuiCardContent-root css-rnega1-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1salclq-MuiTypography-root"
        >
          Frase de prueba 2
        </p>
      </div>
      <button
        aria-label="Eliminar frase"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-2wse5v-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
